GitUI/CommandsDialogs/SubmodulesDialog/FormAddSubmodule.cs

```diff
@@ -70,20 +73,19 @@ private void DirectorySelectedIndexChanged(object sender, EventArgs e)
 
         private void BranchDropDown(object sender, EventArgs e)
         {
-            var module = new GitModule(Directory.Text);
-
-            var heads = new List<IGitRef>
-            {
-                GitRef.NoHead(module)
-            };
-
-            if (module.IsValidGitWorkingDir())
-            {
-                heads.AddRange(module.GetRefs(false));
-            }
-
-            Branch.DisplayMember = nameof(IGitRef.Name);
-            Branch.DataSource = heads;
+            var url = Directory.Text;
+            var gitArguments = new GitArgumentBuilder("ls-remote") { url.ToPosixPath().Quote() };
```

Comment:
Should have --heads, no need to fetch tags too

Reply:
> Should have --heads, no need to fetch tags too

Good point, thank you.
I've kept the check for valid branch refs. On error, the drop down list will be empty.